ayon_server/suggestions/folder.py

```diff
@@ -0,0 +1,123 @@
+from collections import defaultdict
+
+from ayon_server.entities import FolderEntity
+from ayon_server.lib.postgres import Postgres
+
+from .models import (
+    FolderSuggestionItem,
+    TaskSuggestionItem,
+    UserSuggestionItem,
+)
+
+STYPE = list[UserSuggestionItem | TaskSuggestionItem]
+
+
+async def get_folder_suggestions(
+    user: str,
+    folder: FolderEntity,
+) -> dict[str, STYPE]:
+    """
+    Assignees: Every assignee in the project
+    Versions: Disabled - what versions would you want to see on a folder?
+    Tasks: Direct child tasks of the folder.
+    """
+
+    project_name = folder.project_name
+
+    result: defaultdict[str, STYPE] = defaultdict(list)
+
+    # get users:
+
+    query = f"""
+        WITH relevant_users AS (
+            SELECT unnest(t.assignees) as name
+            FROM project_{project_name}.tasks t
+        )
+
+        SELECT
+            u.name as name,
+            u.attrib->>'fullName' as label,
+            r.rel_count as has_task
+        FROM users u
+        LEFT JOIN LATERAL (
+            SELECT count(*) as rel_count
+            FROM project_{project_name}.tasks t
+            JOIN project_{project_name}.hierarchy h ON t.folder_id = h.id
+            WHERE
+                t.assignees @> ARRAY[u.name]
+                AND h.path LIKE '{folder.path.lstrip("/")}%'
```

Comment:
Does passing these strings in directly like this into the query mean we're now prone to SQL injection attacks?

Reply:
project name is sanitized at the API level (in `ayon_server.api.dependencies`) and it won't allow any potentially malicious characters.

folder path is auto-generated as a materialized view from folder names, which also have strict rules on what characters could be used. Without handling queries as strings, I'd be f**** at many different places :)

Postgres does not allow passing table names as query arguments, so specifying schema as part of the query is necessary for all project-scoped FROM clauses.